Mirror/Runtime/NetworkAnimator.cs

```diff
@@ -231,27 +231,29 @@ internal void HandleAnimTriggerMsg(int hash)
 
         void WriteParameters(NetworkWriter writer, bool autoSend)
         {
-            for (int i = 0; i < m_Animator.parameters.Length; i++)
+            // store the animator parameters in a variable - the "Animator.parameters" getter allocates
+            // a new parameter array every time it is accessed so we should avoid doing it in a loop
+            AnimatorControllerParameter[] parameters = m_Animator.parameters;
+
+            for (int i = 0; i < parameters.Length; i++)
             {
                 if (autoSend && !GetParameterAutoSend(i))
                     continue;
 
-                AnimatorControllerParameter par = m_Animator.parameters[i];
+                AnimatorControllerParameter par = parameters[i];
                 if (par.type == AnimatorControllerParameterType.Int)
                 {
                     writer.WritePackedUInt32((uint)m_Animator.GetInteger(par.nameHash));
 
                     SetSendTrackingParam(par.name + ":" + m_Animator.GetInteger(par.nameHash), i);
                 }
-
-                if (par.type == AnimatorControllerParameterType.Float)
+                else if (par.type == AnimatorControllerParameterType.Float)
```

Comment:
sorry to be a pain in the ass, but could you just leave the 'if' parts completely untouched and open another PR for that?^^

Reply:
Sure thing, no problem